upi/vsphere/machine/variables.tf

```diff
@@ -20,6 +20,10 @@ variable "resource_pool_id" {
   type = "string"
 }
 
+variable "folder_id" {
```

Comment:
I'm not fond of `folder_id` as the name of this variable. It is not an ID. It is the name of the folder. My preference would be for this to just be `folder`. I would be fine with `folder_name` or `folder_path`, too.